### PR TITLE
emissive blockers are now just an overlay to lower maptick (proof in comments)

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
currently emissive blockers are objects added to the vis_contents of every item. however mutable appearances as overlays can serve this role perfectly well and according to my tests should cause less maptick per item. for mobs mutable appearances dont work for reasons i dont understand so instead this adds the em_block object to overlays instead of vis_contents. these both use atom/movable/update_icon() now

graphed test results below 

## Why It's Good For The Game
LESS MAPTICK. by some amount at least. i dont know how much of a contribution to maptick the emissive blockers on items + mobs make specifically for a live round. the only downsides to this that i know of (assuming there arent weird bugs) are that init probably takes slightly longer and each item takes more memory (probably) because overlays are value-copied appearances instead of references like vis_contents are

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
gib graphs/proof

**Comment by Kylerace** [comment]
TLDR: in terms of maptick cost, no emissive blockers at all ?< emissive blockers as overlays < emissive blockers as vis_contents (what we have now)

all of these tests were done in this incomplete branch: https://github.com/Kylerace/tgstation/tree/mapticktesting on byond version 
1549 and currently this branch tends to crash randomly after around 400 minutes at minimum (the minimum that ive found so far at least), but beyond cutting some tests shorter than i would otherwise prefer it doesnt seem to affect the result, if its unacceptable i can try repeating these tests on the older branch https://github.com/Kylerace/tgstation/tree/mapticktestingnopr that doesnt crash (assuming the crash isnt due to 514). the files on that branch of interest are maptick.dm (the subsystem that is recording maptick related stats every 0.5 seconds) and tests_objects.dm (where all the objects im using for these tests are defined)

there are two types of test: unmoving and moving. 
the moving test involves an automover that goes up and down at around ~1/2 the speed of normal running because im an idiot but all the tests have it so whatever. in all tests the entire view of the character is taken up by 50 of the object in question per tile (asd tells me the SendMaps turf bubble is 19x21 blocks with widescreen, so thats 19950 objects being sent to the player)

the three categories are: objects with no emissive blockers (called generic objects), objects with vis_contents emissive blockers, and objects with overlays emissive blockers.

**Objects with No Emissive Blockers**
Unmoving:
L-Generic Objects 50l, Average Maptick 7 3771 (crashed after 500 minutes)
Moving:
L-Moving Generic Objects 50l, Average Maptick 12 475 (crashed after 640 minutes)(1)

**Objects with Overlays as Emissive Blockers**
Unmoving:
L-Emissive Blocker as Overlays, Average Maptick 7 33962
L-Emissive Blocker Test 50l With Overlays, Average Maptick 7 23709 (crashed after 700 minutes)
average across both: 7.288355
Moving:
L-Moving Test Emissive Blockers as Overlays, Average Maptick 12 6018 (crashed after 670 minutes)

**Objects with vis_contents Emissive Blockers**
Unmoving:
L-Emissive Blocker Test 50l w_ vis_contents, Average Maptick 12 9549 (crashed after 400 minutes)
L-Emissive Blocker Test Generic Items 50l, Average Maptick 13 5551
Moving:
L-Moving Test Emissive Blockers as vis_contents, Average Maptick 19 0057 (crashed after 590 minutes)

**Comment by Kylerace** [comment]
wow thats a chonker

**Comment by Cyberboss** [comment]
I'll take your word for it
